dash/dash.py

```diff
@@ -990,15 +999,25 @@ def add_context(*args, **kwargs):
                         )
 
                     component_ids = collections.defaultdict(dict)
+                    has_update = False
                     for i, o in enumerate(output):
-                        component_ids[o.component_id][o.component_property] =\
-                            output_value[i]
+                        val = output_value[i]
+                        if val is not no_update:
+                            has_update = True
+                            o_id, o_prop = o.component_id, o.component_property
+                            component_ids[o_id][o_prop] = val
```

Comment:
Glad that we're returning a dict instead of a list here 😸 I was sure that we would've had to change the renderer to do this.